### PR TITLE
Various documentation fixes

### PR DESCRIPTION
**Pull request by gkelly**
This PR includes fixes for several things:
- The logo not rendering in README.md
- The menu using non-underlined text for links
- A change in title from "General Documentation" to "Documentation Index" for the top-level index.
- A minor copy update for the Firmware Update document regarding encrypted update.

### CODE REVIEW

File: doc/security/specs/firmware_update/index.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ The purpose of this document is to capture the process by which Tock will
 perform a complete in-place B-slot update of the running firmware and request
 that the `ROM/ROM_EXT` code boot into it.
 
+Note that this document does not cover the encrypted update process. That is
+left as a detail to be covered by the implmentation of the loader itself.
+
```
Comment (domrizz0) [inline doc/security/specs/firmware_update/index.md:23]: Maybe more clearly something like "Note: for certification reasons this document does not discuss detailed guidance for encrypted updates. Additional implementation details on both regular and encrypted updates to follow when possible."

@moidx for his thoughts

**Review by domrizz0 — CHANGES_REQUESTED** [review]
Left some notes on clarifying wording.

File: doc/security/specs/firmware_update/index.md (hunk 1 above)
Comment (gkelly) [reply doc/security/specs/firmware_update/index.md:24]: Updated copy to your suggestion.

**Review by moidx — APPROVED** [review]
LGTM

**Review by domrizz0 — APPROVED** [review]
lgtm

**Pull request merged**